src/main/java/org/nobloat/bare/dsl/Lexer.java

```diff
@@ -0,0 +1,168 @@
+            case '[':
+                return new Token(Token.Type.L_BRACKET);
+            case ']':
+                return new Token(Token.Type.R_BRACKET);
+            case '(':
+                return new Token(Token.Type.L_PAREN);
+            case ')':
+                return new Token(Token.Type.R_PAREN);
+            case '|':
+                return new Token(Token.Type.PIPE);
+            case '=':
+                return new Token(Token.Type.EQUAL);
+            case ':':
+                return new Token(Token.Type.COLON);
+        }
+
+        return new Token(Token.Type.UNKNOWN);
+    }
+
+    private Token tokenOfWord(String word) {
+
+        switch (word) {
+            case "type":
+                return new Token(Token.Type.TYPE);
+            case "enum":
+                return new Token(Token.Type.ENUM);
+            case "uint":
+                return new Token(Token.Type.UINT);
+            case "u8":
+                return new Token(Token.Type.U8);
+            case "u16":
+                return new Token(Token.Type.U16);
+            case "u32":
+                return new Token(Token.Type.U32);
+            case "u64":
+                return new Token(Token.Type.U64);
+            case "int":
+                return new Token(Token.Type.INT);
+            case "i8":
+                return new Token(Token.Type.I8);
+            case "i16":
+                return new Token(Token.Type.I16);
+            case "i32":
+                return new Token(Token.Type.I32);
+            case "i64":
+                return new Token(Token.Type.I64);
+            case "f32":
+                return new Token(Token.Type.F32);
+            case "f64":
+                return new Token(Token.Type.F64);
+            case "bool":
+                return new Token(Token.Type.BOOL);
+            case "string":
+                return new Token(Token.Type.STRING);
+            case "data":
+                return new Token(Token.Type.DATA);
+            case "void":
+                return new Token(Token.Type.VOID);
+            case "optional":
+                return new Token(Token.Type.OPTIONAL);
+            case "map":
+                return new Token(Token.Type.MAP);
+        }
+
+        return new Token(Token.Type.NAME, word);
+    }
+
+    public static class Token {
+
+        final Type type;
+        final String value;
+
+        public Token(Type type) {
+            this(type, null);
+        }
+
+        public Token(Type type, String value) {
+            this.type = type;
+            this.value = value;
+        }
+
+        enum Type {
+            INTEGER,
+            L_ANGLE, R_ANGLE,
+            L_BRACE, R_BRACE,
+            L_BRACKET, R_BRACKET,
+            L_PAREN, R_PAREN,
+            PIPE,
+            EQUAL,
+            COLON,
+            UNKNOWN,
+            EOF,
+            COMMENT,
+
+            TYPE,
+            ENUM,
+            UINT,
+            U8,
```

Comment:
I think the number types could at least be grouped together.

Reply:
agree, done!